internal/parser/argument_parsers.go

```diff
@@ -22,6 +23,30 @@ func parseSetFunction(command string, arguments []string, cacheMap map[string]*K
 	return keyValueObject
 }
 
+func parseHSetFunction(command string, arguments []string, cacheMap map[string]*KeyValue) *KeyValue {
+	keyValueObject, exists := cacheMap[arguments[0]]
+
+	if !exists {
+		keyValueObject := new(KeyValue)
+		keyValueObject.key = arguments[0]
+		keyValueObject.value = arguments[1:]
+		keyValueObject.command = command
+	
+		return keyValueObject
+	}
+
+	for _, newValue := range arguments[1:] {
+		for _, value := range keyValueObject.value {
```

Comment:
Can you convert the values already in the keyValueObject.value to a map and then overwrite existing fields with values that match them. That way, we achieve two things, the algorithm runs in O(n) time complexity instead of the current O(nm) complexity it is and also we are able to overwrite existing fields with new values. Thank you.

Reply:
But I'll have to flatten the map back to the keyValueObject.value slice type